service/controller/aws/v15/resource/certconfigcleansing/resource.go

```diff
@@ -0,0 +1,107 @@
+package certconfigcleansing
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	Name = "certconfigcleansingv1"
+)
+
+// Config represents the configuration used to create a new certconfigcleansing resource.
+type Config struct {
+	G8sClient versioned.Interface
+	Logger    micrologger.Logger
+}
+
+// Resource implements the certconfigcleansing resource.
+type Resource struct {
+	g8sClient versioned.Interface
+	logger    micrologger.Logger
+}
+
+func (r Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+
+	currentMap := map[string]bool{}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "Collecting all AWSClusterConfigs")
+
+	rs, err := r.g8sClient.CoreV1alpha1().AWSClusterConfigs("").List(metav1.ListOptions{})
```

Comment:
The resource design is totally backwards. Now I see why it would make sense to do this in `cert-operator` instead.

Reply:
Are you suggesting move all the changes to cert-operator? :(

Reply:
I suggest to understand what the "best" solution would be. My personal goal is not to get a problem solved somehow, but in the best possible way.